p2p/transport/webrtc/connection.go

```diff
@@ -0,0 +1,369 @@
+package libp2pwebrtc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"math/rand"
+	"os"
+	"sync"
+	"sync/atomic"
+
+	ic "github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	tpt "github.com/libp2p/go-libp2p/core/transport"
+	pb "github.com/libp2p/go-libp2p/p2p/transport/webrtc/pb"
+	"github.com/libp2p/go-msgio"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/pion/datachannel"
+	"github.com/pion/webrtc/v3"
+	"google.golang.org/protobuf/proto"
+)
+
+var _ tpt.CapableConn = &connection{}
+
+const (
+	maxAcceptQueueLen = 10
+)
+
+type acceptStream struct {
+	stream  datachannel.ReadWriteCloser
+	channel *webrtc.DataChannel
+}
+
+type connection struct {
+	// debug identifier for the connection
+	dbgId     int
+	pc        *webrtc.PeerConnection
+	transport *WebRTCTransport
```

Comment:
@GlenDC For the M3 implementation, could we type this field as `tpt.Transport`. We only have this field for the `connection.Transport` method.

Reply:
not sure what you mean?

Reply:
I want to change the field's type from `*WebRTCTransport` to `tpt.Transport`. It will allow using the connection struct in browser-to-browser.

Reply:
Ok sure, will look into it.

Reply:
Done.

Reply:
Can we please do M3-related changes in a separate PR? This PR is complicated enough as is.

Reply:
@GlenDC Please revert.

Reply:
Sure, seemed like an innocent enough change, but I do get your point. Consider it done @marten-seemann  :)